### PR TITLE
[beta] Support of message update endpoint

### PR DESCRIPTION
**Pull request by symphony-thibault**
Added new message update endpoint to `MessageService` class.
- Mentioned in markdown doc 
- Simple demo created (`MessageUpdateExampleMain.java`). I'd like make it a bit more fun so I'll modify it before merging this PR
----
message-update-show

### CODE REVIEW

File: symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/message/MessageService.java [hunk 1]
```diff
@@ -289,6 +289,26 @@ public V4Message send(@Nonnull String streamId, @Nonnull Message message) {
         () -> this.doSendMessage(streamId, message));
   }
 
+  /**
+   * Update an existing message. The existing message must be a valid social message, that has not been deleted.
+   *
+   * @param streamId the ID of the stream to send the message to
+   * @param messageId the ID of the message to be
+   * @return a {@link V4Message} object containing the details of the sent message
+   * @see <a href="https://developers.symphony.com/restapi/reference#update-message-v4">Create Message v4</a>
```
Comment (symphony-youri) [inline symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/message/MessageService.java:298]: Update Message v4? (in the link's name)

File: symphony-bdk-examples/bdk-core-examples/src/main/java/com/symphony/bdk/examples/MessageUpdateExampleMain.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * This demonstrates a basic usage of the message update feature.
+ */
+public class MessageUpdateExampleMain {
+
+  public static void main(String[] args) throws Exception {
+
+    final SymphonyBdk bdk = new SymphonyBdk(loadFromSymphonyDir("config.yaml"));
+
+    bdk.activities().register(slash("/go", false, context -> {
+      newSingleThreadExecutor().submit(() -> startTheShow(bdk, context.getStreamId()));
```
Comment (symphony-youri) [inline symphony-bdk-examples/bdk-core-examples/src/main/java/com/symphony/bdk/examples/MessageUpdateExampleMain.java:23]: why do we need the thread executor?

**Review by symphony-youri — APPROVED** [review]

**Review by symphony-mariacristina — APPROVED** [review]
Should we maybe also mention that we cannot update a message containing attachments?

**Comment by symphony-thibault** [comment]
> @symphony-mariacristina - Should we maybe also mention that we cannot update a message containing attachments?

Good point, we have different solutions: 
1. throw an error if `Message#attachments` is not empty
2. create another `Message` that does not contain `attachments` and `previews` fields
3. just mention that in Javadoc

I personally prefer solution `3`, as we plan to support attachments in a near future, it will avoid us to modify the BDK once message update works with attachments.

**Comment by symphony-mariacristina** [comment]
> I personally prefer solution `3`, as we plan to support attachments in a near future, it will avoid us to modify the BDK once message update works with attachments.

yes, I agree with point 3. The API is marked as experimental and will be changed anyway to support attachments.

File: symphony-bdk-examples/bdk-core-examples/src/main/java/com/symphony/bdk/examples/MessageUpdateExampleMain.java (hunk 2 above)
Comment (symphony-thibault) [reply symphony-bdk-examples/bdk-core-examples/src/main/java/com/symphony/bdk/examples/MessageUpdateExampleMain.java]: To void blocking the DF loop, and run multiple `/go` at the same time. But I just updated the example and removed it

File: symphony-bdk-examples/bdk-core-examples/src/main/java/com/symphony/bdk/examples/MessageUpdateExampleMain.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+  @SneakyThrows
+  private static void startTheShow(SymphonyBdk bdk, String streamId) {
+
+    // just for this example, obviously!
+    Function<Integer, String> pickEmoji = index -> new String[] {
+        "cat", "dromedary_camel", "dolphin", "dog", "hamster", "goat", "panda_face", "koala", "frog", "penguin"
+    }[index];
+
+    V4Message previousMessage = bdk.messages().send(streamId, "Let's the show begin!");
```
Comment (symphony-thibault) [inline symphony-bdk-examples/bdk-core-examples/src/main/java/com/symphony/bdk/examples/MessageUpdateExampleMain.java:40]:
```suggestion
    V4Message previousMessage = bdk.messages().send(streamId, "Let the show begin!");
```

**Pull request merged**